cmd/atlas/internal/cmdext/cmdext.go

```diff
@@ -441,6 +443,57 @@ func SchemaHCL(ctx *hcl.EvalContext, block *hclsyntax.Block) (cty.Value, error)
 	}), nil
 }
 
+// SchemaExternal is a data source that reads an Atlas SQL/HCL schema state from external program.
+func SchemaExternal(ctx *hcl.EvalContext, block *hclsyntax.Block) (cty.Value, error) {
+	var (
+		args struct {
+			Program []string `hcl:"program"`
+			Dir     string   `hcl:"working_dir,optional"`
+			Remain  hcl.Body `hcl:",remain"`
+		}
+		errorf = blockError("data.external_schema", block)
+	)
+	if diags := gohcl.DecodeBody(block.Body, ctx, &args); diags.HasErrors() {
+		return cty.NilVal, errorf("decoding body: %v", diags)
+	}
+	attrs, diags := args.Remain.JustAttributes()
+	if diags.HasErrors() {
+		return cty.NilVal, errorf("getting attributes: %v", diags)
+	}
+	if len(attrs) > 0 {
+		return cty.NilVal, errorf("unexpected attributes: %v", attrs)
+	}
+	if len(args.Program) == 0 {
+		return cty.NilVal, errorf("program cannot be empty")
+	}
+	cmd := exec.Command(args.Program[0], args.Program[1:]...)
+	if args.Dir != "" {
+		cmd.Dir = args.Dir
+	}
```

Comment:
What about env variables?

Reply:
The default behavior is the environment variables defined in the process. Do you mean updating the PWD if Dir was set?